src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -632,36 +647,37 @@ private static bool ShouldUseKnownLength(BoundCollectionExpression node, out int
             return false;
         }
 
-        /// <summary>
-        /// Create and populate an array from a collection expression where the
-        /// collection has a known length, although possibly including spreads.
-        /// </summary>
-        private BoundExpression CreateAndPopulateArray(BoundCollectionExpression node, ArrayTypeSymbol arrayType)
+        /// <summary>Attempt to optimize conversion of a single-spread collection expr to array, even if the spread length is not known.</summary>
+        private BoundExpression? TryOptimizeSingleSpreadToArray(BoundCollectionExpression node, ArrayTypeSymbol arrayType)
```

Comment:
There's now 3 possible optimizations this method will try, in order:

1. List.ToArray if the spread value is a list
2. IEnumerable.ToArray if our checks say we should convert the spread value in this way
3. ReadOnlySpan.ToArray if we can convert the spread value to ReadOnlySpan

In my opinion this is the best order to try these, in part because IEnumerable.ToArray tends to produce less IL than the span conversions.

Reply:
Done